src/groupeddataframe/grouping.jl

```diff
@@ -366,9 +378,33 @@ _nrow(x::NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}) =
 _ncol(df::AbstractDataFrame) = ncol(df)
 _ncol(x::Union{NamedTuple, DataFrameRow}) = length(x)
 
-function gen_fun(::Type{NT}, f2) where NT
-    get_val(::Val{T}) where {T} = T
+struct Reduce{F, A}
+    f::F
+    adjust::A
+end
+Reduce(f) = Reduce(f, nothing)
+function (r::Reduce)(x)
+    v = reduce(r.f, x)
+    r.adjust === nothing ? v : r.adjust(v, length(x))
+end
+
+check_reduction(f::Any) = f
+check_reduction(::typeof(sum)) = Reduce(Base.add_sum)
+check_reduction(::typeof(prod)) = Reduce(Base.mul_prod)
+check_reduction(::typeof(maximum)) = Reduce(max)
+check_reduction(::typeof(minimum)) = Reduce(min)
+check_reduction(::typeof(mean)) = Reduce(Base.add_sum, /)
 
+for f in (:sum, :prod, :maximum, :minimum, :mean)
+    @eval begin
+        funname(::typeof(check_reduction($f))) = Symbol($f)
```

Comment:
maybe a more specific method name than `funname` could be used?

Reply:
That's the name of the method we use elsewhere already. What's wrong with it?

Reply:
Why not use `nameof` directly here?

Reply:
Because we want the name of the original function, which isn't stored anywhere in the struct. Also what `funname` currently does is return `function` for anonymous functions (which you recently improved unless I'm mistaken).